--- gridlink_old.py	2019-04-09 10:51:50.149275759 -0500
+++ gridlink_new.py	2019-04-09 10:55:13.013274339 -0500
@@ -33,6 +33,10 @@
     from hfk import *
 except:
     TkHFK=None
+try:
+    from transHFK import *
+except:
+    Tk_transHFK=None
 
 revision = '$Revision: 1.75 $'.replace('$','')
 
@@ -201,10 +205,12 @@
         movemenu.add_separator()
         movemenu.add_command(label='Simplify', command=self.simplify)
         menubar.add_cascade(label='Moves', menu=movemenu)
+        invariantmenu = Menu(menubar, tearoff=0)
         if (TkHFK):
-            invariantmenu = Menu(menubar, tearoff=0)
             invariantmenu.add_command(label='HFK^', command=self.HFKhat)
-            menubar.add_cascade(label='Invariants', menu=invariantmenu)
+        if (Tk_transHFK):
+            invariantmenu.add_command(label="transverseHFK", command=self.transHFK)
+        menubar.add_cascade(label='Invariants', menu=invariantmenu)
         menubar.add_cascade(label='Windows', menu=self.app.windowmenu)
         menubar.add_cascade(label='Help', menu=self.app.helpmenu)
         self.window.config(menu=menubar)
@@ -1078,6 +1084,21 @@
         hfk_object = TkHFK(Xlist, Olist, name=self.window.title())
         hfk_object.HFK_ranks()
 
+    def transHFK(self):
+        if self.components > 1:
+            showwarning('Knots only',
+                        'Sorry, I can only compute transverseHFK invariants for knots')
+            return
+        self.reflect()
+        self.reflect()
+        self.reflect()
+        Xlist, Olist = self.get_XOlists()
+        self.reflect()
+        if 0 in Xlist:
+            Xlist = [x+1 for x in Xlist]
+            Olist = [o+1 for o in Olist]
+        t_hfk_object = Tk_transHFK(nXlist, nOlist, name=self.window.title(), parent=self)
+
     def winding_numbers(self):
         result = []
         for i in range(self.size):
